src/rpc/net.cpp

```diff
@@ -282,6 +283,61 @@ static UniValue addnode(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static UniValue addconnection(const JSONRPCRequest& request)
+{
+    const RPCHelpMan help{"addconnection",
+        "\nOpen an outbound connection to a specified node (test only)",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The IP address and port to attempt connecting to."},
+            {"connectiontype", RPCArg::Type::STR, RPCArg::Optional::NO, "'outbound' or 'blockrelay'"},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                { RPCResult::Type::STR, "address", "Address of newly added connection." },
+                { RPCResult::Type::STR, "connectiontype", "Type of connection opened." },
+            }},
+        RPCExamples{
+            HelpExampleCli("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+            + HelpExampleRpc("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+        },
+    };
+
+    help.Check(request);
+
+    if (!Params().IsMockableChain()) {
+        throw std::runtime_error("addconnection is for regression testing (-regtest mode) only");
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VSTR});
+    const std::string address = request.params[0].get_str();
+    const std::string conn_type_in = request.params[1].get_str();
+    ConnectionType conn_type{};
+    if (conn_type_in == "outbound") {
+        conn_type = ConnectionType::OUTBOUND;
+    } else if (conn_type_in == "blockrelay") {
+        conn_type = ConnectionType::BLOCK_RELAY;
+    } else {
+        throw std::runtime_error(help.ToString());
+    }
+
+    NodeContext& context = EnsureNodeContext(request.context);
+    if (!context.connman) {
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+    }
+
+    bool success = context.connman->AddConnection(address, conn_type);
```

Comment:
yeah, I know I could inline but I separated it into its own variable in an attempt to emphasize that this is where the actual work happens since it gets compiled out anyways. so I don't think the `const` does anything, but I threw it in there anyways